chain/chunks/src/client.rs

```diff
@@ -242,8 +240,8 @@ mod tests {
 
         tracing::info!("checking the pool after resharding");
         {
-            let num_shards = new_shard_layout.num_shards();
-            for shard_id in 0..num_shards {
+            let shard_ids: Vec<_> = new_shard_layout.shard_ids().collect();
```

Comment:
Nit: collect_vec from itertools is an alternative to explicit type. It's just preference, up to you.